src/libman/Commands/CacheCleanCommand.cs

```diff
@@ -64,6 +65,13 @@ protected override Task<int> ExecuteInternalAsync()
                     Logger.Log(string.Format(Resources.Text.CacheCleanFailed, ex.Message), LogLevel.Error);
                 }
             }
+            else
+            {
+                if (!ManifestDependencies.Providers.Any(p => p.Id == Provider.Value))
```

Comment:
This should just be an `else if`.

Reply:
For future reference, I had chopped this down because there's still a case that doesn't output a message -- if the directory doesn't exist but it _is_ a valid provider.  I couldn't find a suitable message in the existing translations, so leaving this out (this case isn't mentioned in the original issue anyway).